### ISSUE TITLE
TemplateError in postgres.yaml

### ISSUE THREAD

**Issue by sethrosenblum** [DataDog/chef-datadog#300]
Seeing this error when running chef-client on a host with the postgres check enabled:
```
      ================================================================================
      Error executing action `create` on resource 'template[/etc/dd-agent/conf.d/postgres.yaml]'
      ================================================================================
      
      Chef::Mixin::Template::TemplateError
      ------------------------------------
      Undefined node attribute or method `keys?' on `node'
      
      Resource Declaration:
      ---------------------
      suppressed sensitive resource output
      
      Compiled Resource:
      ------------------
      suppressed sensitive resource output
      
      Template Context:
      -----------------
      on line #28
       26:         <% end -%>
       27:     <% end -%>
       28:     <% if i.keys?("collect_function_metrics") -%>
       29:     collect_function_metrics: i["collect_function_metrics"]
       30:     <% end -%>
```

This looks to be a result of #278 .  The `i` object here is an [ImmutableHash](http://www.rubydoc.info/github/opscode/chef/Chef/Node/ImmutableMash) so we probably want to run `i.key?('collect_function_metrics')` here.

**Comment by miketheman** [comment]
See proposed fix in #302. This looks to apply the desired change.

**Comment by sethrosenblum** [comment]
This is a showstopper for anyone using the postgres check.  Any chance we could hotfix this out while waiting for someone to rewrite spec tests?

**Comment by miketheman** [comment]
@sethrosenblum Please review #304 - it fixes another bug in the parameter wrapped in the `if` in the first place.

**Comment by miketheman** [comment]
@sethrosenblum Released in 2.4.0
